app/controllers/item/additionalReference/AddAnotherAdditionalReferenceController.scala

```diff
@@ -53,23 +57,33 @@ class AddAnotherAdditionalReferenceController @Inject() (
         case 0 =>
           Redirect(controllers.item.routes.AddAdditionalReferenceYesNoController.onPageLoad(lrn, mode, itemIndex))
         case _ =>
-          Ok(view(form(viewModel), lrn, viewModel, itemIndex))
+          val preparedForm = request.userAnswers.get(AddAnotherAdditionalReferencePage(itemIndex)) match {
+            case None        => form(viewModel)
+            case Some(value) => form(viewModel).fill(value)
+          }
+          Ok(view(preparedForm, lrn, viewModel, itemIndex))
       }
   }
 
-  def onSubmit(lrn: LocalReferenceNumber, mode: Mode, itemIndex: Index): Action[AnyContent] = actions.requireData(lrn) {
+  def onSubmit(lrn: LocalReferenceNumber, mode: Mode, itemIndex: Index): Action[AnyContent] = actions.requireData(lrn).async {
     implicit request =>
       val viewModel = viewModelProvider(request.userAnswers, mode, itemIndex)
       form(viewModel)
         .bindFromRequest()
         .fold(
-          formWithErrors => BadRequest(view(formWithErrors, lrn, viewModel, itemIndex)),
-          {
-            case true =>
-              Redirect(controllers.item.additionalReference.index.routes.AdditionalReferenceController.onPageLoad(lrn, mode, itemIndex, viewModel.nextIndex))
-            case false =>
-              Redirect(navigatorProvider(mode, itemIndex).nextPage(request.userAnswers, Some(AdditionalReferencesSection(itemIndex))))
-          }
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, viewModel, itemIndex))),
+          value =>
+            AddAnotherAdditionalReferencePage(itemIndex)
+              .writeToUserAnswers(value)
+              .updateTask()
+              .writeToSession(sessionRepository)
+              .navigateTo {
+                if value then
+                  controllers.item.additionalReference.index.routes.AdditionalReferenceController.onPageLoad(lrn, mode, itemIndex, viewModel.nextIndex)
+                else navigatorProvider(mode, itemIndex).nextPage(request.userAnswers, Some(AddAnotherAdditionalReferencePage(itemIndex)))
```

Comment:
For complicated navigation reasons this needs to remain `AdditionalReferencesSection` in the current page parameter. I'll have a think over how we can make this a bit nicer as on the face of it `AddAnotherAdditionalReferencePage` and `AdditionalReferencesSection` are inherently linked